corehq/apps/app_manager/tasks.py

```diff
@@ -23,26 +22,40 @@ def create_user_cases(domain_name):
 
 
 @serial_task('{app_id}-{version}', max_retries=0, timeout=60 * 60)
-def make_async_build_v2(app_id, domain, version, allow_prune=True, comment=None):
+def make_async_build_v2(app_id, domain, version, username, allow_prune=True, comment=None):
     app = get_app(domain, app_id)
-    return make_async_build(app, allow_prune=allow_prune, comment=comment)
-
-
-@serial_task('{app._id}-{app.version}', max_retries=0, timeout=60 * 60)
-def make_async_build(app, allow_prune=True, comment=None):
-    previous_version = app.get_previous_version()
-    if previous_version and previous_version.version == app.version:
-        return
     errors = app.validate_app()
     if not errors:
         if allow_prune and comment is None:
             comment = _('Auto-generated by a phone update. Will expire after next build if not marked released.')
+        if username:
+            comment = (comment + ' ') or ''
+            comment += _('Generated by user {}.').format(username)
         copy = app.make_build(comment=comment)
         copy.is_auto_generated = allow_prune
         copy.save(increment_version=False)
         return copy
 
 
+def make_async_build(app, username, allow_prune=True, comment=None):
```

Comment:
I think this is now only ever called with `app, username`, so you can safely drop `allow_prune` and `comment`

Reply:
https://github.com/dimagi/commcare-hq/pull/24860/commits/ab12e720288565b18e665793d4fa4eb8aaca5004